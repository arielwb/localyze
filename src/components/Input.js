import React from 'react'

export const Input = ({ type, value, label, name, onSubmit, onChange }) => {
  const handleSubmit = (event) => {
    event.preventDefault()
    onSubmit && onSubmit(event)
  }

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor={name}>{label}</label>
      <input data-testid='input' name={name} type={type} onChange={onChange} value={value} />
      {onSubmit && <button type='submit' data-testid='submit-button'>confirm</button>}
    </form>
  )
}
