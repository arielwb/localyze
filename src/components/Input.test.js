import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, screen } from '@testing-library/react'
import { Input } from './Input'

describe('Input', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(<Input type='text' value='a value' label='A test' name='test' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call the given onSubmit function', () => {
    const onSubmitSpy = jest.fn()
    render(
      <Input
        type='text'
        label='A test'
        name='test'
        onSubmit={onSubmitSpy}
      />
    )
    const button = screen.getByTestId('submit-button')

    fireEvent.submit(button)

    expect(onSubmitSpy).toHaveBeenCalled()
  })

  it('should call the given onChange function', () => {
    const onChangeSpy = jest.fn()
    render(
      <Input
        type='text'
        value='initial value'
        label='A test'
        name='test'
        onChange={onChangeSpy}
      />
    )
    const input = screen.getByTestId('input')
    fireEvent.change(input, { target: { value: 'new value' } })

    expect(onChangeSpy).toHaveBeenCalled()
  })
})
