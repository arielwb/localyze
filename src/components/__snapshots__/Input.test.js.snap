// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="test"
  >
    A test
  </label>
  <input
    data-testid="input"
    name="test"
    type="text"
    value="a value"
  />
</form>
`;
